index.html

```diff
@@ -972,64 +972,127 @@ <h2>UTF-8 Character Encoding Form</h2>
+	</details>
+</div>
+
+<p>Visual rendering introduces the notion of a <a href="https://www.w3.org/TR/i18n-glossary/#def_glyph">glyph</a>. Glyphs are defined by [[ISO/IEC 9541-1]] as <q>a recognizable abstract graphic symbol which is independent of a specific design</q>. There is <em>not</em> a one-to-one correspondence between characters and glyphs:</p>
+<ul>
+<li>A single character can be represented by multiple glyphs
+(each glyph is then part of the representation of that character). These glyphs
+may be physically separated from one another. </li>
+
+<li>A single glyph may represent a sequence of characters (this
+is the case with ligatures, among others).</li><li>A character may be rendered with very different glyphs depending on the context.</li>
+
+<li>A single glyph may represent different characters (e.g.
+capital Latin A, capital Greek A and capital Cyrillic A).</li>
+</ul>
+
+<p>A set of glyphs makes up a <span class="new-term">font</span>. Glyphs can be
+construed as the basic units of organization of the visual rendering of text,
+just as characters are the basic unit of organization of encoded text.</p>
+
+<p>See <a href="https://www.w3.org/TR/charmod/#sec-CharExamples">Examples of Characters, Keystrokes and Glyphs</a> for examples of the complexities of character to glyph mapping.</p>
+
 
 <div class="req" id="char_keystroke">
 	<p class="advisement">Specifications and software MUST NOT require nor depend on a single keystroke resulting in a single character, nor that a single character be input with a single keystroke (even with modifiers), nor that keyboards are the same all over the world.</p>
-	<details class="links"><summary>explanations &amp; examples</summary>
+	<details class="links"><summary>source</summary>
 	<p><a href="https://www.w3.org/TR/charmod/#sec-InputUnits">Units of input C005</a>, in <cite>Character Model for the World Wide Web 1.0: Fundamentals</cite>.</p>
 	</details>
-	</div>
+	<details class="links"><summary>explanations &amp; examples</summary>
+	<p><a href="https://www.w3.org/TR/charmod/#sec-CharExamples">Examples of Characters, Keystrokes and Glyphs</a>, in <cite>Character Model for the World Wide Web 1.0: Fundamentals</cite>.</p>
+	</details>
+  </div>
+
+<p>In keyboard input, it is not always the case that keystrokes and input characters correspond one-to-one. A limited number of keys can fit on a keyboard. Some keyboards will generate multiple characters from a single keypress. In other cases ('<span class="qterm">dead keys</span>') a key will generate no characters, but affect the results of subsequent keypresses. Many writing systems have far too many characters to fit on a keyboard and must rely on more complex <span class="new-term">input methods</span>, which transform keystroke sequences into character sequences. Other languages may make it necessary to input some characters with special modifier keys.</p>
```

Comment:
should `input method` (and IME) be a term in the glossary?

Reply:
Probably, yes.